python-avd/pyavd/_eos_designs/structured_config/overlay/utils.py

```diff
@@ -119,66 +79,9 @@ def _is_peer_mpls_client(self: AvdStructuredConfigOverlayProtocol, peer_facts: d
     def _is_peer_mpls_server(self: AvdStructuredConfigOverlayProtocol, peer_facts: dict) -> bool:
         return peer_facts.get("mpls_overlay_role") == "server" or (peer_facts.get("evpn_role") == "server" and get(peer_facts, "overlay.evpn_mpls") is True)
 
-    @cached_property
-    def _ipvpn_gateway_remote_peers(self: AvdStructuredConfigOverlayProtocol) -> dict:
-        if self.shared_utils.overlay_ipvpn_gateway is not True:
-            return {}
-
-        ipvpn_gateway_remote_peers = {}
-
-        for remote_peer in self.shared_utils.node_config.ipvpn_gateway.remote_peers._natural_sorted():
-            # These remote gw are outside of the inventory
-
-            bgp_as = remote_peer.bgp_as
-
-            ipvpn_gateway_remote_peers[remote_peer.hostname] = {
-                "bgp_as": str(bgp_as) if bgp_as is not None else None,
-                "ip_address": remote_peer.ip_address,
-                # Not adding the "overlay_peering_interface" since we do not know it for this device. Only used for description.
-            }
-
-        return ipvpn_gateway_remote_peers
-
-    @cached_property
-    def _mpls_route_clients(self: AvdStructuredConfigOverlayProtocol) -> dict:
-        if self._is_mpls_server is not True:
-            return {}
-
-        mpls_route_clients = {}
-
-        for avd_peer in self._avd_overlay_peers:
-            peer_facts = self.shared_utils.get_peer_facts(avd_peer, required=True)
-            if self._is_peer_mpls_client(peer_facts) is not True:
-                continue
-
-            if self.shared_utils.hostname in peer_facts.get("mpls_route_reflectors", []) and avd_peer not in self._mpls_route_reflectors:
-                self._append_peer(mpls_route_clients, avd_peer, peer_facts)
-
-        return mpls_route_clients
-
-    @cached_property
-    def _mpls_mesh_pe(self: AvdStructuredConfigOverlayProtocol) -> dict:
-        if not self.shared_utils.overlay_mpls or not self.inputs.bgp_mesh_pes:
-            return {}
-
-        mpls_mesh_pe = {}
-        for fabric_switch in self.shared_utils.all_fabric_devices:
-            if self._mpls_route_reflectors is not None and fabric_switch in self._mpls_route_reflectors:
-                continue
-            if fabric_switch == self.shared_utils.hostname:
-                continue
-
-            peer_facts = self.shared_utils.get_peer_facts(fabric_switch, required=True)
-            if self._is_peer_mpls_client(peer_facts) is not True:
-                continue
-
-            self._append_peer(mpls_mesh_pe, fabric_switch, peer_facts)
-
-        return mpls_mesh_pe
-
     @cached_property
     def _mpls_route_reflectors(self: AvdStructuredConfigOverlayProtocol) -> dict:
-        if self._is_mpls_client is not True:
+        if (self.shared_utils.mpls_overlay_role == "client" or (self.shared_utils.evpn_role == "client" and self.shared_utils.overlay_evpn_mpls)) is not True:
```

Comment:
lets see if we cannot get rid of the dict once we have addressed all comments (probably not)

Reply:
We can handle this in the facts PR afterwards